src/Compilers/CSharp/Portable/Parser/LexerCache.cs

```diff
@@ -181,21 +183,28 @@ internal bool TryGetKeywordKind(string key, out SyntaxKind kind)
             return kind != SyntaxKind.None;
         }
 
-        internal SyntaxTrivia LookupTrivia<TArg>(
-            ReadOnlySpan<char> textBuffer,
-            int hashCode,
-            Func<TArg, SyntaxTrivia> createTriviaFunction,
-            TArg data)
+        internal SyntaxTrivia LookupWhitespaceTrivia(
+            in SlidingTextWindow textWindow,
+            int lexemeStartPosition,
+            int hashCode)
         {
-            var value = TriviaMap.FindItem(textBuffer, hashCode);
+            var span = TextSpan.FromBounds(lexemeStartPosition, textWindow.Position);
+            Debug.Assert(span.Length > 0);
 
-            if (value == null)
+            if (textWindow.TryGetTextIfWithinWindow(span, out var lexemeTextSpan))
```

Comment:
note the user of spans here now, which is really nice for having the text window pass out the span of characters of interest, which then is exactly what the TriviaMap operates on.